src/test/regress/expected/union_pushdown.out

```diff
@@ -937,18 +1055,394 @@ SELECT SUM(k) FROM v;
+                            UNION ALL
+                            SELECT user_id
+                            FROM   cte_2
+                            UNION
+                            SELECT *
+                            FROM   (
+                                            SELECT   user_id
+                                            FROM     cte_2
+                                            GROUP BY user_id) AS bar
+                            UNION
+                            SELECT user_id
+                            FROM   cte_3 ) AS subqu) AS bar
+using  (user_id)
+WHERE  (
+              foo.user_id) IN
+                               (
+                               SELECT user_id
+                               FROM            (
+                                                      SELECT foo.user_id
+                                                      FROM   users_table_part u1
+                                                      JOIN   lateral
+                                                             (
+                                                                    SELECT u1.user_id
+                                                                    FROM   users_table_part u2
+                                                                    WHERE  u2.user_id = u1.user_id) AS foo
+                                                      ON     (
+                                                                    true)
+                                                      UNION
+                                                      SELECT foo.user_id
+                                                      FROM   users_table_part u1
+                                                      JOIN   lateral
+                                                             (
+                                                                    SELECT u1.user_id
+                                                                    FROM   users_table_part u2
+                                                                    WHERE  u2.user_id = u1.user_id) AS foo
+                                                      ON     (
+                                                                    true) ) AS bar
+                               UNION
+                               SELECT *
+                               FROM   cte_2
+                               WHERE  user_id IN
+                                      (
+                                             SELECT user_id
+                                             FROM   users_table_part)
+                               UNION
+                               SELECT *
+                               FROM   cte_1 ) ;
+$$);
+ explain_has_distributed_subplan
+---------------------------------------------------------------------
+ f
+(1 row)
+
+SELECT public.explain_has_distributed_subplan($$
+EXPLAIN SELECT * FROM users_table_part u1 WHERE (value_1, user_id) IN
+(
```

Comment:
Do you know why this fails? For example, if we only had `user_id` on the WHERE clause, would it pushdown?

Reply:
https://github.com/citusdata/citus/blob/f00c63c33daf3d16f06462626ca14732b141ae7a/src/backend/distributed/planner/relation_restriction_equivalence.c#L235

I think we are hitting this, maybe I can add that as a comment.